src/worker/tasks/nonceResyncWorker.ts

```diff
@@ -0,0 +1,103 @@
+import { Job, Processor, Worker } from "bullmq";
+import { eth_getTransactionCount, getRpcClient } from "thirdweb";
+import {
+  lastUsedNonceKey,
+  recycleNonce,
+  splitSentNoncesKey,
+} from "../../db/wallets/walletNonce";
+import { getConfig } from "../../utils/cache/getConfig";
+import { getChain } from "../../utils/chain";
+import { logger } from "../../utils/logger";
+import { redis } from "../../utils/redis/redis";
+import { thirdwebClient } from "../../utils/sdk";
+import { NonceResyncQueue } from "../queues/nonceResyncQueue";
+import { logWorkerExceptions } from "../queues/queues";
+
+// Must be explicitly called for the worker to run on this host.
+export const initNonceResyncWorker = async () => {
+  const config = await getConfig();
+  if (config.minedTxListenerCronSchedule) {
+    NonceResyncQueue.q.add("cron", "", {
+      repeat: { pattern: config.minedTxListenerCronSchedule },
+      jobId: "nonce-resync-cron",
+    });
+  }
+
+  const _worker = new Worker(NonceResyncQueue.q.name, handler, {
+    connection: redis,
+    concurrency: 1,
+  });
+  logWorkerExceptions(_worker);
+};
+
+export const sleep = async (ms: number) =>
+  new Promise((resolve) => setTimeout(resolve, ms, null));
+
+/**
+ * Resyncs nonces for all wallets.
+ * This worker should be run periodically to ensure that nonces are not skipped.
+ * It checks the onchain nonce for each wallet and recycles any missing nonces.
+ */
+const handler: Processor<any, void, string> = async (job: Job<string>) => {
+  const sentNoncesKeys = await redis.keys("*nonce-sent*");
+  job.log(`Found ${sentNoncesKeys.length} nonce-sent* keys`);
+
+  for (const sentNonceKey of sentNoncesKeys) {
+    const { chainId, walletAddress } = splitSentNoncesKey(sentNonceKey);
+
+    // Check blockchain for nonce
+    const rpcRequest = getRpcClient({
+      client: thirdwebClient,
+      chain: await getChain(chainId),
+    });
+
+    // The next unused nonce = transactionCount.
+    const transactionCount = await eth_getTransactionCount(rpcRequest, {
+      address: walletAddress,
+    });
+
+    // Get DB Nonce
+    const dbNonceCount = Number(
+      await redis.get(lastUsedNonceKey(chainId, walletAddress)),
+    );
```

Comment:
There's an existing helper func for this: `inspectNonce(...)`

Also these var names seem confusing and this is a very easy thing to mix up:
- The transactionCount is the _number of transactions sent from this wallet_. 5 means this wallet has sent 5 transactions, the most recent being nonce 4.
- The last used nonce Redis key stores the _last used nonce value for a wallet_. 5 means this wallet already sent nonce 5 and has sent 6 transactions.

We shouldn't compare the two values since the former is the "next unused nonce" and the latter is the "last used nonce".

To make it more clear, let's try to consistently name them and document them. Let's pick last used nonce.
- `lastUsedNonceOnchain = transactionCount - 1`
- `lastUsedNonceDB = inspectNonce(...)`